### PR TITLE
Hookify DatePicker, DropZone, and Filters

### PR DESCRIPTION
**Pull request by AndrewMusgrave**
### WHY are these changes introduced?

Convert examples to hooks

### WHAT is this pull request doing?

Converting DatePicker, DropZone, and Filters class-based examples to hooks

### STILL TO DO

I'm making quite a few pr's converting examples to hooks so I anticipate lots of `unreleased.md` conflicts, so I'll add the entry before I merge.

```
- Converted `DatePicker`, `DropZone`, and `Filters` examples to functional components ([#2129](https://github.com/Shopify/polaris-react/pull/2129))
```

### How to 🎩

Make sure the behavior of the example mirrors the style guide 

</details>

### 🎩 checklist

* [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [ ] Updated the component's `README.md` with documentation changes
* [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
* [ ] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the [Polaris UI kit](https://polaris.shopify.com/resources/polaris-ui-kit)

### CODE REVIEW

File: src/components/DatePicker/README.md [hunk 1]
```diff
@@ -48,40 +48,32 @@ Date pickers should:
...
+    start: new Date('Wed Feb 07 2018 00:00:00 GMT-0500 (EST)'),
+    end: new Date('Wed Feb 07 2018 00:00:00 GMT-0500 (EST)'),
+  });
+
+  const handleMonthChange = useCallback(
+    (month, year) => setDate({month, year}),
+    [],
+  );
+
+  const {month, year} = date;
```
Comment (BPScott) [inline src/components/DatePicker/README.md]: You could  do this in the existing destructuring above

```
const [{month, year}, setDate] = useState({
```

File: src/components/DropZone/README.md [hunk 2]
```diff
@@ -142,69 +139,63 @@ Use to pair with a label for better accessibility.
...
-        title="The following images couldn’t be uploaded:"
-        status="critical"
+  return (
+    <Stack vertical>
+      {errorMessage}
+      <DropZone
+        accept="image/*"
+        type="image"
+        onDrop={(files, acceptedFiles, rejectedFiles) => {
+          setFiles([...files, ...acceptedFiles]);
```
Comment (BPScott) [inline src/components/DropZone/README.md]: Should we make this a callback like in the prior example?

File: src/components/DropZone/README.md [hunk 3]
```diff
@@ -142,69 +139,63 @@ Use to pair with a label for better accessibility.
...
-              source={window.URL.createObjectURL(file)}
-            />
-            <div>
-              {file.name} <Caption>{file.size} bytes</Caption>
-            </div>
-          </Stack>
+function DropZoneWithImageFileUpload() {
+  const [files, setFiles] = useState([]);
+  const [rejectedFiles, setRejectedFiles] = useState([]);
+  const [hasError, setHasError] = useState(false);
```
Comment (BPScott) [inline src/components/DropZone/README.md]: Do we need a state to track this? It seems to be derived from other state info so I think we'd be able to do  `const hasError = rejectedFiles.length > 0`

File: src/components/DropZone/README.md [hunk 4]
```diff
@@ -276,68 +263,65 @@ class DropZoneExample extends React.Component {
...
-              source={window.URL.createObjectURL(file)}
-            />
-            <div>
-              {file.name} <Caption>{file.size} bytes</Caption>
-            </div>
-          </Stack>
+function DropZoneAcceptingSVGFilesExample() {
+  const [files, setFiles] = useState([]);
+  const [rejectedFiles, setRejectedFiles] = useState([]);
+  const [hasError, setHasError] = useState(false);
```
Comment (BPScott) [inline src/components/DropZone/README.md]: As above, do we need a state to track this?

File: src/components/DropZone/README.md [hunk 5]
```diff
@@ -346,53 +330,50 @@ class DropZoneExample extends React.Component {
...
-          </DropZone>
-        </Card>
-      </DropZone>
-    );
-  }
+function NestedDropZoneExample() {
+  const [files, setFiles] = useState([]);
+
+  const handleDropZoneDrop = useCallback(
+    (dropFiles, _acceptedFiles, _rejectedFiles) =>
```
Comment (BPScott) [inline src/components/DropZone/README.md]: If we never use `_acceptedFiles` or `_rejectedFiles` do we need to mention them? There's no other parameter after them that we need

File: src/components/DropZone/README.md [hunk 6]
```diff
@@ -425,64 +406,63 @@ Use for cases with tight space constraints, such as variant thumbnails on the Pr
...
+  const [files, setFiles] = useState([]);
+  const [openFileDialog, setOpenFileDialog] = useState(false);
+
+  const handleDropZoneDrop = useCallback(
+    (dropFiles, _acceptedFiles, _rejectedFiles) =>
+      setFiles([...files, ...dropFiles]),
+    [files],
+  );
+  const toggleOpenFileDialog = useCallback(
+    () => setOpenFileDialog(!openFileDialog),
```
Comment (BPScott) [inline src/components/DropZone/README.md]: use callback style of `setOpenFileDialog`

File: src/components/DropZone/README.md (hunk 5 above)
Comment (AndrewMusgrave) [reply src/components/DropZone/README.md]: I think it'd be nice to keep them in, so consumers know the parameters are available without skimming the docs

**Comment by AndrewMusgrave** [comment]
Thanks for the suggestions @BPScott 😄  , I was only converting the examples but it looks like there were quite a few quick wins, not sure why hasError was on state 🤷‍♂

**Pull request merged**